docs/source/chapt_pyomo/tutorial/pyomo_test.rst

```diff
@@ -23,51 +23,44 @@ Instructions
 4. Add x1, x2, & y as “Output Variables” within the Node Editor (They are meant to correspond to the decision variable values of the optimization model).
 
-5. Under Node Script, set Script Mode to “Post”, and enter the following PYOMO code for the optimization model:
-
-  * ``from pyomo.environ import * (1)``
-  * ``from pyomo.opt import SolverFactory (2)``
-  * ``import pyutilib.subprocess.GlobalData (3)``
-  * ``pyutilib.subprocess.GlobalData.DEFINE_SIGNAL_HANDLERS_DEFAULT = False (4)``
-  * ``m=ConcreteModel() (5)``
-  * ``m.x1=Var(within=PositiveReals) (6)``
-  * ``m.x2=Var(within=PositiveReals) (7)``
-  * ``m.y=Var() (8)``
-  * ``m.c1=Constraint(expr=x["a"]*m.x1+x["b"]*m.x2>=x["c"]) (9)``
-  * ``m.c2=Constraint(expr=m.x1+m.x2==m.y) (10)``
-  * ``m.o=Objective(expr=m.y) (11)``
-  * ``opt = SolverFactory("gams") (12)``
-  * ``io_options = dict() (13)``
-  * ``io_options['solver'] = 'cplex' (14)``
-  * ``io_options['mtype'] = 'mip' (15)``
-  * ``opt.solve(m,io_options=io_options) (16)``
-  * ``f["x1"]=m.x1.value (17)``
-  * ``f["x2"]=m.x2.value (18)``
-  * ``f["y"]=m.y.value (19)``
+5. Under Node Script, set Script Mode to “Post”. This will ensure that the node script runs after the node simulation.
+   Enter the following PYOMO code for the optimization model:
+   code:: python
+
+  * ``1. from pyomo.environ import (Var,Constraint,ConcreteModel,value,Param)``
+  * ``2. from pyomo.opt import SolverFactory``
+  * ``3. import pyutilib.subprocess.GlobalData ``
+  * ``4. pyutilib.subprocess.GlobalData.DEFINE_SIGNAL_HANDLERS_DEFAULT = False ``
+  * ``5. m=ConcreteModel() ``
+  * ``6. m.x1=Var(within=PositiveReals) ``
+  * ``7. m.x2=Var(within=PositiveReals) ``
+  * ``8. m.y=Var() ``
+  * ``9. m.c1=Constraint(expr=x["a"]*m.x1+x["b"]*m.x2>=x["c"]) ``
+  * ``10.m.c2=Constraint(expr=m.x1+m.x2==m.y) ``
+  * ``11.m.o=Objective(expr=m.y)``
+  * ``12.opt = SolverFactory("ipopt") ``
+  * ``13.opt.solve(m) ``
+  * ``14.f["x1"]=m.x1.value ``
+  * ``15.f["x2"]=m.x2.value ``
+  * ``16.f["y"]=m.y.value ``
 
 In the above code, lines (1), (2) are used to import the PYOMO package and SolverFactory function to develop the model and solve it by accessing an appropriate solver.
 
-The solvers can be accessed in 2 ways:
-(i) PYOMO Solver Plugins: directly accessing the solvers available within Pyomo, like glpk, cplex, baron, etc.
-
-Eg: Opt = SolverFactory(“glpk”).
-
-(ii) PYOMO GAMS Interface: In case some of the licensed solvers like BARON or CPLEX are not directly available to users, they can be accessed in PYOMO through GAMS.
+A PYOMO Concrete Model is declared, defining the variables, declaring the constraints using the parameters defined within “Input Variables” of the Node, and defining the objective function. (Lines (5) to (11)).
 
-More information on the syntax can be found in the following link: https://pyomo.readthedocs.io/en/latest/library_reference/solvers/gams.html.
+Lines (12) & (13) are used to access the solver, and solve the model via ipopt solver.
```

Comment:
Line 12 sets the solver to ipopt and line 13 sends the problem to be solved. Ipopt is a nonlinear optimization solver.